### PR TITLE
Migrate `xpallets-assets` to attribute macro

### PR DESCRIPTION
**Pull request by 0xrjman**
# Migrate XAssets Record

## Migrate Guide

1. Subtrate issue[#7882](https://github.com/paritytech/substrate/issues/7882)
2. Chainx issue[#552](https://github.com/chainx-org/ChainX/issues/522)
3. Upgrade-[guidelines](https://crates.parity.io/frame_support/attr.pallet.html#upgrade-guidelines)

## Migrate Description

migrate xpallet-assets `module` to `pallet`(attribute macro)

### Check Upgrade Status -> `Pass`

1. Check the format of pallet(template required)
pallet-format
2. Use `subsee` to do a diff of metadata(origin, after migration)
   - add 1 comment
     - add-1
     - add-2
   - amend 3 comments
     - amend-1
     - amend-2

### CODE REVIEW

File: xpallets/assets/src/lib.rs [hunk 1]
```diff
@@ -237,41 +158,172 @@ decl_module! {
...
+        Issued(AssetId, T::AccountId, BalanceOf<T>),
+        /// Some balances of an asset were destoryed. [asset_id, who, amount]
+        Destroyed(AssetId, T::AccountId, BalanceOf<T>),
+        /// Set asset balance of an account by root. [asset_id, who, asset_type, amount]
+        BalanceSet(AssetId, T::AccountId, AssetType, BalanceOf<T>),
+    }
+
+    /// Old name generated by `decl_event`.
+    #[deprecated(note = "use `Event` instead")]
+    pub type RawEvent<T> = Event<T>;
```
Comment (hacpy) [inline xpallets/assets/src/lib.rs]: There is no ``RawEvent`` here, this is unnecessary.

File: xpallets/assets/src/lib.rs [hunk 2]
```diff
@@ -24,210 +24,131 @@ pub mod weights;
...
             Self::can_transfer(&id)?;
-            Self::move_usable_balance(&id, &transactor, &dest, value).map_err::<Error::<T>, _>(Into::into)?;
+            Self::move_usable_balance(&id, &transactor, &dest, value)
+                .map_err::<Error<T>, _>(Into::into)?;
             Ok(())
         }
 
         /// set free token for an account
-        #[weight = 0]
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
```
Comment (hacpy) [inline xpallets/assets/src/lib.rs]:
```suggestion
        #[pallet::weight(0)]
```
Why not use 0 directly as before?

File: xpallets/assets/src/lib.rs [hunk 3]
```diff
@@ -237,41 +158,172 @@ decl_module! {
             Ok(())
         }
 
-        #[weight = <T as Config>::WeightInfo::set_asset_limit()]
-        pub fn set_asset_limit(origin, #[compact] id: AssetId, restrictions: AssetRestrictions) -> DispatchResult {
+        /// asset restriction method reserved for root
+        #[pallet::weight(<T as Config>::WeightInfo::zero())]
```
Comment (hacpy) [inline xpallets/assets/src/lib.rs]:
```suggestion
        #[pallet::weight(<T as Config>::WeightInfo::set_asset_limit())]
```
Is there any other reason to set it to 0？

File: xpallets/gateway/common/src/mock.rs [hunk 4]
```diff
@@ -14,6 +14,7 @@ use sp_runtime::{
     AccountId32, DispatchError, DispatchResult,
 };
 
+use crate::mock::sp_api_hidden_includes_construct_runtime::hidden_include::traits::GenesisBuild;
```
Comment (hacpy) [inline xpallets/gateway/common/src/mock.rs]: Better to use ``frame_support::traits::GenesisBuild;``

File: xpallets/assets/src/lib.rs [hunk 5]
```diff
@@ -510,10 +554,9 @@ impl<T: Config> Module<T> {
         }
         Ok(())
     }
-}
 
-/// token issue destroy reserve/unreserve, it's core function
-impl<T: Config> Module<T> {
+    /// Token issue destroy reserve/unreserved, it's core function
+    ///
```
Comment (liuchengxu) [inline xpallets/assets/src/lib.rs]:
```suggestion
```

File: xpallets/assets/src/lib.rs [hunk 6]
```diff
@@ -343,10 +391,9 @@ impl<T: Config> Module<T> {
         }
         Ok(())
     }
-}
 
-// Public read functions.
-impl<T: Config> Module<T> {
+    /// Public read functions.
+    ///
```
Comment (liuchengxu) [inline xpallets/assets/src/lib.rs]:
```suggestion
    // Public read functions.

```

File: xpallets/assets/src/lib.rs [hunk 7]
```diff
@@ -237,41 +158,169 @@ decl_module! {
...
             Error::<T>::DenyNativeAsset
         );
         Ok(())
     }
-}
 
-// asset related
-impl<T: Config> Module<T> {
+    /// Asset related
+    ///
```
Comment (liuchengxu) [inline xpallets/assets/src/lib.rs]:
```suggestion
    // Asset related
    
```

**Pull request merged**